deepchem/models/tensorgraph/layers.py

```diff
@@ -808,18 +858,27 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
     if in_layers is None:
       in_layers = self.in_layers
     in_layers = convert_to_layers(in_layers)
+
+    try:
+      shape = self.shape
+    except NotImplementedError:
```

Comment:
@peastman This is one of the places where I had to modify the behavior. See the original self._shape references. I'm not sure if subclasses of Layer are supposed to access via the private members.

Reply:
Okay! I'll keep this as is then.